pkg/controllers/node/termination/suite_test.go

```diff
@@ -129,34 +143,43 @@ var _ = Describe("Termination", func() {
 				nodes = append(nodes, node)
 			}
 
-			var wg sync.WaitGroup
-			// this is enough to trip the race detector
-			for i := 0; i < 10; i++ {
-				wg.Add(1)
-				go func(node *v1.Node) {
-					defer GinkgoRecover()
-					defer wg.Done()
-					ExpectObjectReconciled(ctx, env.Client, terminationController, node)
-				}(nodes[i])
-			}
-			wg.Wait()
-			ExpectNotFound(ctx, env.Client, lo.Map(nodes, func(n *v1.Node, _ int) client.Object { return n })...)
+			Eventually(func(g Gomega) {
+				var wg sync.WaitGroup
+				// this is enough to trip the race detector
+				for i := 0; i < numNodes; i++ {
+					wg.Add(1)
+					go func(node *v1.Node) {
+						defer GinkgoRecover()
+						defer wg.Done()
+						_, err := reconcile.AsReconciler(env.Client, terminationController).Reconcile(ctx, reconcile.Request{NamespacedName: client.ObjectKeyFromObject(node)})
+						g.Expect(err).ToNot(HaveOccurred())
+					}(nodes[i])
+				}
+				wg.Wait()
+				for _, n := range nodes {
+					tmpNode := &v1.Node{}
+					g.Expect(errors.IsNotFound(env.Client.Get(ctx, client.ObjectKeyFromObject(n), tmpNode))).To(BeTrue())
+				}
+				ExpectNotFound(ctx, env.Client, lo.Map(nodes, func(n *v1.Node, _ int) client.Object { return n })...)
```

Comment:
Aren't we validating the same thing on line 161?

Reply:
Oh that's a miss on my part, will remove. This is actually a no-op since we can't use these expectations inside `Eventually`.